python/tests/test_lowlevel.py

```diff
@@ -1496,6 +1496,13 @@ def test_time_units(self):
             ts.load_tables(tables)
             assert ts.get_time_units() == value
 
+    def test_extend_edges_bad_args(self):
+        ts1 = self.get_example_tree_sequence(10)
+        with pytest.raises(TypeError):
+            ts1.extend_edges()
+        with pytest.raises(TypeError, match="an integer"):
+            ts1.extend_edges("sdf")
```

Comment:
Need to add a test for ``max_iter=0`` here (or something else that will trigger a C-level error). Otherwise error handling code won't come up as covered.